eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version1/ChannelCodecs1.scala

```diff
@@ -62,7 +62,7 @@ private[channel] object ChannelCodecs1 {
         ("toSelfDelay" | cltvExpiryDelta) ::
         ("maxAcceptedHtlcs" | uint16) ::
         ("isInitiator" | bool8) ::
-        ("defaultFinalScriptPubKey" | lengthDelimited(bytes)) ::
+        ("defaultFinalScriptPubKey" | optional(provide(true), lengthDelimited(bytes))) ::
```

Comment:
```suggestion
        ("upfrontShutdownScript_opt" | lengthDelimited(bytes).map(Option(_)).decodeOnly) ::
```

Reply:
I've used `xmap` here instead of map because we have tests that also need to encode data, but I'm not sure that we need these tests any more ?

Reply:
> I'm not sure that we need these tests any more ?

Those tests then need to be updated, like we did when switching codecs from v3 to v4.